### PR TITLE
handle updating manifests across multiple directories

### PR DESCRIPTION
**Pull request by jakecoffman**
follow-on to #8309

This makes the Updater run multiple updates across multiple directories, then smashes all the changes together and creates a PR. Confirmed with some of the smoke tests: https://github.com/dependabot/smoke-tests/pull/135

### CODE REVIEW

**Comment by jakecoffman** [comment]
Look at https://github.com/dependabot/dependabot-core/pull/8331/commits/d84cd5cc61943f5e6f6c6432458e52313c2fd4e3 before I refactored to more easily see what I changed.

File: updater/lib/dependabot/updater/dependency_group_change_batch.rb [hunk 1]
```diff
@@ -22,9 +22,9 @@ def initialize(initial_dependency_files:)
       end
 
       # Returns an array of DependencyFile objects for the current state
-      def current_dependency_files
-        @dependency_file_batch.map do |_path, data|
-          data[:file]
+      def current_dependency_files(directory)
+        @dependency_file_batch.filter_map do |path, data|
+          data[:file] if path.start_with?(directory)
```
Comment (jakecoffman) [inline updater/lib/dependabot/updater/dependency_group_change_batch.rb:27]: FYI This is not correct, need to do use proper path classes to figure out if the two are exactly the same, not `start_with`.

File: updater/lib/dependabot/updater/operations/create_group_security_update_pull_request.rb [hunk 2]
```diff
@@ -81,6 +63,42 @@ def perform
...
+
+          if job.source.directories.nil?
+            @dependency_change = compile_all_dependency_changes_for(group)
+          else
+            dependency_changes = job.source.directories.map do |directory|
+              job.source.directory = directory
+              # Fixes not updating because it already updated in a previous group
+              dependency_snapshot.handled_dependencies.clear
+              compile_all_dependency_changes_for(group)
+            end
```
Comment (Nishnha) [inline updater/lib/dependabot/updater/operations/create_group_security_update_pull_request.rb:94]: This is a cool approach! I think it makes a lot of sense. I do think we should make the logic of swapping the job directory more explicit though. I'll take a stab at refactoring it

File: updater/lib/dependabot/updater/dependency_group_change_batch.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply updater/lib/dependabot/updater/dependency_group_change_batch.rb:27]: So, if I'm understanding this correctly, the input `dependency_files` to `FileParser`, `UpdateChecker` and `FileUpdater` won't change, correct? In other words, we can still assume that list of files has the same `directory` for all of them?

Comment (jakecoffman) [reply updater/lib/dependabot/updater/dependency_group_change_batch.rb:27]: Correct! 🎯

File: updater/lib/dependabot/updater/operations/create_group_security_update_pull_request.rb (hunk 2 above)
Comment (jakecoffman) [reply updater/lib/dependabot/updater/operations/create_group_security_update_pull_request.rb:94]: I hear you, but I wouldn't worry too much about refactoring and polishing yet. We can do that in follow-on PRs. For now I think it's important to get this out and tested so we can make progress!

**Comment by jakecoffman** [comment]
Our test fixtures had "/bundler" but then we were mocking the files out to be at directory "/" so I made those consistent to get the tests passing.

**Review by honeyankit — APPROVED** [review]
LGTM

**Comment by jakecoffman** [comment]
Upon deploying I noticed an increase in errors which was caused by how security updates come in with directory `/.`. Since this wasn't being normalized it wasn't matching in the new filtering, and so Dependabot was throwing RuntimeError. 

I fixed this in https://github.com/dependabot/dependabot-core/pull/8331/commits/bc259c23958ec80731bbb2837f9d1505ffaaaf58 and added tests.

**Review by honeyankit — APPROVED** [review]

**Comment by jakecoffman** [comment]
Looks like there's one more bug in the same area. When we process `github-actions` jobs we allow directory to be `/` even though the files are at `/.github/workflows` so by filtering out we're causing the same RuntimeError as above. We'll need to add an exception for Actions I think?

The reason our smoke tests didn't catch this is we added a separate directory `/actions` to avoid the smoke test checking our actual workflows for updates. However this kind of setup is extremely rare, so I'll start by making the smoke test for Actions more realistic.

**Comment by jakecoffman** [comment]
I updated the smoke test in https://github.com/dependabot/smoke-tests/pull/139 and now it's correctly failing.

**Comment by jakecoffman** [comment]
A little unfortunate but I had to add a special case in https://github.com/dependabot/dependabot-core/pull/8331/commits/0a4ac826aeeac49edd66974b2bc59832f5c1069b for github-actions where directory `/` actually means `.github/workflows` which is what that ecosystem does.

**Pull request merged**